cpp/src/arrow/acero/bloom_filter_test.cc

```diff
@@ -171,8 +172,12 @@ void TestBloomSmallHashHelper(int64_t num_input_hashes, const T* input_hashes,
 // Output FPR and build and probe cost.
 //
 void TestBloomSmall(BloomFilterBuildStrategy strategy, int64_t num_build,
-                    int num_build_copies, bool use_simd, bool enable_prefetch) {
-  int64_t hardware_flags = use_simd ? ::arrow::internal::CpuInfo::AVX2 : 0;
+                    int num_build_copies, bool use_avx2, bool enable_prefetch) {
+  int64_t hardware_flags = use_avx2 ? CpuInfo::AVX2 : 0;
+  if (hardware_flags && !CpuInfo::GetInstance()->IsSupported(hardware_flags)) {
+    // What else?
+    return;
```

Comment:
This is a parameterized test.  So either we need to change the parameterization in `TEST(BloomFilter, Basic)` to never set `use_avx2` to true if it isn't supported (in which case we can remove this if check) or we can just do `GTEST_SKIP() << "AVX2 not supported on CPU"`.

Reply:
You're right! And this led me to notice that this case was never hit:
https://github.com/apache/arrow/blob/8624a311eb66c32d1b221cfb604cde738e65a210/cpp/src/arrow/acero/bloom_filter_test.cc#L479

(who had the tremendous idea of using a C array with hardcoded size here?)